pallets/portfolio/src/lib.rs

```diff
@@ -329,4 +349,127 @@ impl<T: Trait> Module<T> {
         ensure!(name_uniq, Error::<T>::PortfolioNameAlreadyInUse);
         Ok(())
     }
+
+    /// Accept custody of a Portfolio
+    fn accept_portfolio_custody(new_custodian: IdentityId, auth_id: u64) -> DispatchResult {
+        ensure!(
+            <identity::Authorizations<T>>::contains_key(Signatory::from(new_custodian), auth_id),
+            AuthorizationError::Invalid
+        );
+
+        let auth = <identity::Authorizations<T>>::get(Signatory::from(new_custodian), auth_id);
+
+        let (portfolio_owner, portfolio_number) = match auth.authorization_data {
+            AuthorizationData::PortfolioCustody(did, number) => (did, number),
+            _ => return Err(Error::<T>::IrrelevantAuthorization.into()),
+        };
+
+        let portfolio_id = Self::get_portfolio_id(portfolio_owner, portfolio_number);
+
+        if let Some(num) = portfolio_number {
+            ensure!(
+                !<Portfolios>::contains_key(portfolio_owner, num),
```

Comment:
Intentional `!` here?

I'd introduce:
```rust
fn ensure_has_portfolio(did, num) -> DispatchResult {
    ensure!(<Portfolios>::contains_key(&did, &num), Error::<T>::PortfolioDoesNotExist);
}
```
and use it in the various places `PortfolioDoesNotExist` is raised so intentions are clearer and future changes are easier.